tools/taskcluster/tasks.py

```diff
@@ -111,14 +108,65 @@ def _craft_shell_task(self, name, script, scopes, artifacts):
                 'artifacts': artifacts,
                 'features': {
                     'taskclusterProxy': True,
+                    'chainOfTrust': chain_of_trust,
                 }
+            }
+        })
+
+    def craft_sign_for_github_task(self, build_task_id, is_staging):
+        return self._craft_base_task('Sign for Github', {
+            'provisionerId': 'scriptworker-prov-v1',
+            'workerType': 'mobile-signing-dep-v1' if is_staging else 'mobile-signing-v1',
```

Comment:
nit: `mobile-signing-dep-v1` - did you mean `mobile-signing-dev-v1`? Or maybe `mobile-signing-debug-v1`? I'm not sure what dep is.

Also, since both of these strings need to get updated when the version changes, I wonder if string format args or storing the version in a variable would be more robust than an if expression (though I guess they're maybe less readable in python)

Reply:
Heh, I've been stockholm syndrome'd into forgetting how opaque some of these terms and values are.

"dep" is an old releng term referring to a kind of staging/testing work. It has a relation to the old "depend vs clobber" days, which I'm unfamiliar with.
That worker (`mobile-signing-v1` or `mobile-signing-dep-v1`) is very long lived. When updates happen to the signing software, we update the software on that worker (the "version" in the name does not change). The `v1` in the name probably exists in case we make a radical new signing worker that is completely incompatible with the old worker, and we need a way to differentiate.

Since this happens infrequently, hard-coding the `v1` is effective.